src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs

```diff
@@ -246,7 +249,7 @@ private ServiceProviderEngine GetEngine()
 #if NETFRAMEWORK || NETSTANDARD2_0
             engine = CreateDynamicEngine();
 #else
-            if (RuntimeFeature.IsDynamicCodeCompiled)
+            if (RuntimeFeature.IsDynamicCodeCompiled && !DisableDynamicEngine)
```

Comment:
For other reviewers: the check for `IsDynamicCodeCompiled` here was added prior to v8.